### PR TITLE
fluentd parser plugin to validate OL events

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
### Problem

Experimental support for Fluent proxy. 

Closes: #1728

### Solution

Please describe your change as it relates to the problem, or bug fix, as well as any dependencies. If your change requires a schema change, please describe the schema modifcation(s) and whether it's a _backwards-incompatible_ or _backwards-compatible_ change, then select one of the following:

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: proxy/fluentd/events/event_no_run_id.json
Comment (fm100) [inline proxy/fluentd/events/event_no_run_id.json:1]: The tricky part when I was testing json schema validator with spec was that the relative refs for facets are invalid and the schema validator can't follow through it: https://github.com/OpenLineage/OpenLineage/issues/1440

Depending on the library, some throw an error and some just simply ignore it, but validating the json inside the facet didn't work well, for example something like invalid run UUID format from `parentRun` facet. You may add some test cases validating facets too.

**Review by merobi-hub — APPROVED** [review]
The Fluentd readme looks good to me. Thanks!

File: proxy/fluentd/README.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+## Deployment on Kubernetes
+
+***Section under construction***
+
+## Parser plugin
+
+Openlineage-parser is a Fluentd plugin that verifies if a JSON matches the OpenLineage schema.
+
+### Configuration
```
Comment (mobuchowski) [inline proxy/fluentd/README.md:95]: Can we have also single config to disable all validation altogether?

File: proxy/fluentd/docker/init-db.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#
+# Copyright 2018-2023 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+#
+# Usage: $ ./init-db.sh
+
+set -eu
+
+psql -v ON_ERROR_STOP=1 --username "${POSTGRES_USER}" > /dev/null <<-EOSQL
+  CREATE USER ${MARQUEZ_USER};
```
Comment (mobuchowski) [inline proxy/fluentd/docker/init-db.sh:11]: Why use Marquez related env variables in proxy?

File: proxy/fluentd/README.md (hunk 1 above)
Comment (mobuchowski) [reply proxy/fluentd/README.md:95]: Or, to make sure, that only structure of base `RunEvent` is looked at

Comment (pawel-big-lebowski) [reply proxy/fluentd/README.md:95]: To disable all validation, one should remove the parser from pipeline config. 
To validate `RunEvent` base structure only, all the facets' validations need to be disabled.

**Review by JDarDagran — APPROVED** [review]
Amazing work @pawel-big-lebowski 🚀 🥇

**Pull request merged**